app.py

```diff
@@ -17,9 +17,18 @@ def home() -> str:
     return render_template('home.html')
 
 
+@app.route('/health')
+def health():
+    return "Ok", 200
+
+
 @app.route('/animals', methods=['GET'])
 def index() -> Response:
-    animals = Animal.query.all()
+    search = request.args.get('search')
+    if search:
+        animals = Animal.query.filter(Animal.name.contains(search))
```

Comment:
better to use `icontains` to make search case insensetive

Reply:
Changed